### PR TITLE
Route53 service and cloudformation resources

### PR DESCRIPTION
**Pull request by sjones4**
Tech preview implementation for route53 simple hosted zones plus associated cloudformation resources.

Fixes corymbia/eucalyptus#178

### CODE REVIEW

**Comment by sjones4** [comment]
Build: /job/eucalyptus-internal-5-build-random-rpms/149/
Stage: /job/eucalyptus-internal-5-stage-random/124/
Deploy: /job/eucalyptus-5-ansible-deploy/100/
Test: /job/eucalyptus-5-qa-fast/22/

Demo is that the related route53 tests pass when route53 is enabled / registered:

```
#
# cat /etc/eucalyptus/eucalyptus.conf | grep OPTS
CLOUD_OPTS="-Denable.sqs.tech.preview=true -Denable.route53.tech.preview=true -Xmx2g --bind-addr=10.117.111.14"
#
#
# euserv-register-service -t route53 -h 10.117.111.14 -z api.10.117.111.14 api.10.117.111.14.route53
#
#
# ./gradlew -Dclcip=10.117.111.14 -Dtest.filter=Route53Suite test
...
#
#
# ./gradlew -Dclcip=10.117.111.14 -Dtest.filter=TestCFTemplatesShort test
...
#
#
# ./gradlew -Dclcip=10.117.111.14 -Dtest.filter=TestCFTemplatesFull test
...
#
```

Related test pull request Corymbia/n4j#14

File: clc/modules/route53/src/main/java/com/eucalyptus/route53/service/dns/Route53DnsProperties.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+public class Route53DnsProperties {
+  private static Logger LOG = Logger.getLogger(Route53DnsProperties.class);
+
+  @ConfigurableField(
+      description = "Enable the route53 dns resolver.  Note: dns.enabled must also be 'true'",
+      initial = "true" )
+  public static Boolean RESOLVER_ENABLED = Boolean.TRUE;
+
+  @ConfigurableField(
+      description = "Enable the route53 dns resolver for users hosted zones.  Must also be enabled.",
```
[the remark this answers is not in the document]
Comment (sjones4) [reply clc/modules/route53/src/main/java/com/eucalyptus/route53/service/dns/Route53DnsProperties.java]: Right, this refers to the `resolver_enabled` property also being true:

```
services.route53.dns.resolver_enabled = true
services.route53.dns.user_resolver_enabled = true
```

File: clc/modules/route53/src/main/java/com/eucalyptus/route53/service/Route53Service.java [hunk 2]
```diff
@@ -168,32 +679,187 @@ public ListHealthChecksResponseType listHealthChecks(final ListHealthChecksType
...
+        ctx.getUserFullName( ).asAccountFullName( );
+
+    try {
+      final String canonicalRequestName = Route53DnsHelper.absoluteName(request.getDNSName());
+      final String hostedZoneMatch = request.getHostedZoneId() == null ?
+          null :
+          "/hostedzone/" + request.getHostedZoneId();
+      final Predicate<HostedZone> firstZonePredicate = zone -> {
+        boolean matches = zone != null;
+        if ( matches && canonicalRequestName!=null && !canonicalRequestName.equals(zone.getName()) ) {
```
[the remark this answers is not in the document]
Comment (sjones4) [reply clc/modules/route53/src/main/java/com/eucalyptus/route53/service/Route53Service.java]: That would be more consistent. I'm not going to update it, but for future reference you can suggest this type of change in a review comment.

https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/reviewing-proposed-changes-in-a-pull-request

File: clc/modules/route53/src/main/java/com/eucalyptus/route53/service/dns/Route53DnsProperties.java (hunk 1 above)
Comment (obino) [inline clc/modules/route53/src/main/java/com/eucalyptus/route53/service/dns/Route53DnsProperties.java]:
```suggestion
      description = "Enable the route53 dns resolver for users hosted zones.  Note: dns.resolver_enabled must also be enabled.",
```

**Pull request merged**